hrms/hr/doctype/leave_application/leave_application.js

```diff
@@ -171,10 +171,24 @@ frappe.ui.form.on("Leave Application", {
 		frm.trigger("calculate_total_days");
 	},
 
-	validate_from_to_date: function (frm, null_date) {
+	validate_from_to_date: function (frm, updated_field) {
+		if (!frm.doc.from_date || !frm.doc.to_date) return;
+
 		const from_date = Date.parse(frm.doc.from_date);
 		const to_date = Date.parse(frm.doc.to_date);
-		if (to_date < from_date) frm.set_value(null_date, "");
+
+		if (to_date < from_date) {
+			const other_field = updated_field === "from_date" ? "to_date" : "from_date";
+
+			frm.set_value(other_field, frm.doc[updated_field]);
+			frappe.show_alert({
+				message: __("Changing '{0}' to {1}.", [
+					__(frm.fields_dict[other_field].df.label),
+					frm.doc[updated_field],
```

Comment:
```suggestion
					frappe.datetime.str_to_user(frm.doc[updated_field]),
```
This should format date as per system settings

Reply:
![image](https://github.com/user-attachments/assets/0c36bc25-75c0-40e2-bdad-2dd2f62902c5)

Good catch 😄